src/context-selection/org-unit-selector-bar-item/use-offline-levels.js

```diff
@@ -0,0 +1,102 @@
+import { useQueries } from '@tanstack/react-query'
+
+export const computeDeepestLevel = (
+    rootLevel,
+    offlineLevels,
+    configOfflineLevel
+) => {
+    if (offlineLevels) {
+        return rootLevel + offlineLevels - 1
+    }
+    if (configOfflineLevel) {
+        return configOfflineLevel
+    }
+    return 1
+}
+
+export const computeOfflineLevels = (
+    userOrgUnitRoot,
+    filledOrganisationUnitLevels,
+    configOfflineOrgUnitLevel
+) => {
+    const filledOrganisationUnitLevel = filledOrganisationUnitLevels.find(
+        ({ level }) => level === userOrgUnitRoot.level
+    )
+    const deepestLevel = computeDeepestLevel(
+        userOrgUnitRoot.level,
+        filledOrganisationUnitLevel?.offlineLevels,
+        configOfflineOrgUnitLevel
+    )
+
+    return Array.from(
+        { length: deepestLevel - userOrgUnitRoot.level + 1 },
+        (_, i) => userOrgUnitRoot.level + i
+    )
+}
+
+export default function useOfflineLevels() {
+    const results = useQueries({
+        queries: [
+            {
+                queryKey: [
+                    'me',
+                    {
+                        params: {
+                            fields: ['organisationUnits[id,level,path]'],
+                        },
+                    },
+                ],
+                select: (data) =>
+                    data.organisationUnits.map(({ id, level }) => ({
+                        id,
+                        level,
+                    })),
+            },
+            {
+                queryKey: [
+                    'filledOrganisationUnitLevels',
+                    {
+                        params: {
+                            // @TODO: api ignores the `fields` query param
+                            // See: https://jira.dhis2.org/browse/TECH-973
+                            fields: ['level', 'offlineLevels'],
+                        },
+                    },
+                ],
+                select: (data) =>
+                    data.map(({ id, level, offlineLevels }) => ({
+                        id,
+                        level,
+                        offlineLevels,
+                    })),
+            },
+            {
+                queryKey: ['configuration/offlineOrganisationUnitLevel'],
+                select: ({ level }) => level,
+            },
+        ],
```

Comment:
This "violates" the app's standard how to construct query keys.
See [here](https://github.com/dhis2/aggregate-data-entry-app/blob/development/src/data-workspace/query-key-factory.js) or [here](https://github.com/dhis2/aggregate-data-entry-app/blob/development/src/shared/use-data-value-set/query-key-factory.js) for examples.